app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,121 @@
+class RecipesController < ApplicationController
+  include ActionView::Helpers::NumberHelper
+
+  before_action :set_recipe, only: %i[show edit update destroy generate_shopping_list]
+
+  def index
+    @recipes = Recipe.all
+  end
+
+  def show
+    @recipe = Recipe.includes(:foods).find(params[:id])
+    @foods = @recipe.foods
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.build(recipe_params)
+
+    if @recipe.save
+      redirect_to @recipe, notice: 'Recipe was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def public_recipes
+    # @public_recipes = Recipe.where(public: true)
+    @public_recipes = Recipe.public_recipes.where.not(user: current_user)
+  end
+
+  def edit; end
+
+  def update
+    if @recipe.update(recipe_params)
+      redirect_to @recipe, notice: 'Recipe was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @recipe = current_user.recipes.find(params[:id])
+
+    if @recipe.destroy
+      redirect_to recipes_path, notice: 'Recipe deleted successfully'
+    else
+      redirect_to recipes_path, notice: 'Recipe cannot delete'
+    end
+  end
+
+  def update_public_status
+    @recipe = Recipe.find(params[:id])
+    @recipe.update(public: params[:public])
+    render json: { status: 'success' }
+  end
+
+  def toggle_public_status
+    @recipe = Recipe.find(params[:id])
+    @recipe.toggle_public_status_and_visibility
+
+    redirect_to @recipe, notice: 'Recipe public status updated successfully.'
+  end
```

Comment:
- [ ] There is a liitle adjsutment you need to make for public_recipe items; I tried to create a recipe as `public` when I wanted to add a **new recipe** and when I **toggled the button** on the `recipe_show` page but the recipe were not added to the `public_recipe` page.